src/main/java/org/jenkinsci/plugins/workflow/graph/StandardGraphLookupView.java

```diff
@@ -33,9 +30,22 @@
     /** Map a node to its nearest enclosing block */
     HashMap<String, String> nearestEnclosingBlock = new HashMap<String, String>();
 
+    /** Map a block start node to its immediate children nodes, including AtomNode, BlockStartNode, and BlockEndNode */
+    HashMap<String, List<String>> blockChildren = new HashMap<>();
+
     public void clearCache() {
         blockStartToEnd.clear();
         nearestEnclosingBlock.clear();
+        blockChildren.clear();
+    }
+
+    private void addToBlockChildren(@Nonnull String blockId, @Nonnull String childId) {
```

Comment:
If we just directly expose an iterable/iterator over children, no need for this.